### ISSUE TITLE
Roaring64Map support

### ISSUE THREAD

**Issue by stuarteberg** [Ezibenroc/PyRoaringBitMap#58]
The CRoaring source code includes a [C++ file named `roaring64map.hh`][1], which begins with the following comment:

[1]: https://github.com/RoaringBitmap/CRoaring/blob/master/cpp/roaring64map.hh

```
/*
A C++ header for 64-bit Roaring Bitmaps, implemented by way of a map of many
32-bit Roaring Bitmaps.
*/
```

So... do you think it would be possible to add python bindings for `Roaring64Map`?  Or is that not feasible (or not advisable) for some reason?

**Comment by stuarteberg** [comment]
BTW (unrelated): I added a [`pyroaring` recipe to conda-forge][1], so users can now install it with the following command:

[1]: https://github.com/conda-forge/pyroaring-feedstock

```
conda install -c conda-forge pyroaring
```

Perhaps that should be mentioned as an installation option in the README?

**Comment by Ezibenroc** [comment]
Hello, and sorry for the delay.

It would be nice to have bindings for `Roaring64Map`. It would be quite a large amount of work though and I do not have much time presently, but if you want to give it a try, go for it!

The conda installation is pretty cool, thanks. Would you want to do a pull request for the README to phrase it yourself?

**Comment by stuarteberg** [comment]
Thanks for the response.  I don't know if I need 64-bit support after all :-)

>Would you want to do a pull request for the README to phrase it yourself?

See #61

I've also opened https://github.com/conda-forge/pyroaring-feedstock/pull/4 to add you as a maintainer to the conda-forge package.  I'll merge it if you approve.

**Comment by Ezibenroc** [comment]
> Thanks for the response. I don't know if I need 64-bit support after all :-)

Alright, I will keep this issue open anyway, in case someone else wants to try ;-)

**Comment by yihuang** [comment]
I tried to wrap the c++ interface of roaring64map, but struggle to get it build, because I can't set flag `-std=c++20` to cpp files only.
But then I created a pure python implementation: https://github.com/yihuang/python-roaring64, wrapping PyRoaringBitMap bitmap as inner containers.

**Comment by lemire** [comment]
You should not need C++20 to build CRoaring.

**Comment by yihuang** [comment]
> You should not need C++20 to build CRoaring.

it output some compile error which could be fixed with that flag though, I don't have the exact error message at hand now.

**Comment by Ezibenroc** [comment]
An alternative implementation of the 64 bit bitmaps is available in CRoaring, see https://github.com/RoaringBitmap/CRoaring/pull/534
It should be more efficient than the existing one (caveat: I don't think it has been benchmarked yet).
It is also maybe easier to wrap since it is written in C.
Note that we still depend on this issue being closed if we want to use this new implementation: https://github.com/RoaringBitmap/CRoaring/issues/543

**Comment by lemire** [comment]
https://github.com/RoaringBitmap/CRoaring/issues/543 has been solved just now.

As soon as we have serialization/deserialization, I will issue a new release.

**Comment by Ezibenroc** [comment]
Fixed in the latest release [1.0.0](https://github.com/Ezibenroc/PyRoaringBitMap/releases/tag/1.0.0) 🎉
